### PR TITLE
PS-8560 [8.0]: Allow to build RocksDB for selected CPU architecture

### PR DESCRIPTION
**Pull request by inikep**
1. Introduce `ROCKSDB_BUILD_ARCH` parameter to select MyRocks build architecture with `-march=${ROCKSDB_BUILD_ARCH}`.
2. Remove adding instruction set flags e.g. `-msse4.2` to individual files.

### CODE REVIEW

File: storage/rocksdb/CMakeLists.txt [hunk 1]
```diff
@@ -161,32 +167,37 @@ ENDIF()
...
-      MESSAGE(FATAL_ERROR "No SSE42 support found. Not building MyRocks")
+      SET(ROCKSDB_BUILD_ARCH "native")
     ENDIF()
   ENDIF()
+
+  # CMAKE_CXX_FLAGS are used by CHECK_CXX_SOURCE_RUNS called from ROCKSDB_SET_DEFINTIONS()
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=${ROCKSDB_BUILD_ARCH}")
+  MESSAGE(STATUS "MyRocks build architecture: -march=${ROCKSDB_BUILD_ARCH}")
+  ROCKSDB_SET_X86_DEFINTIONS()
 ELSE() # aarch64
```
Comment (percona-ysorokin) [inline storage/rocksdb/CMakeLists.txt:204]: Shouldn't we do something similar to the `AARCH64` branch as well?
I believe `ROCKSDB_DISABLE_MARCH_NATIVE` can be reused here as well
and minimal `ROCKSDB_BUILD_ARCH` can be set to, say, `arvv8-a`

File: storage/rocksdb/CMakeLists.txt [hunk 2]
```diff
@@ -161,51 +167,58 @@ ENDIF()
...
+    IF (ROCKSDB_DISABLE_MARCH_NATIVE)
+      # Intel Westmere is oldest CPU that supports SSE 4.2 and PCLMUL instruction sets
+      SET(ROCKSDB_BUILD_ARCH "westmere")
     ELSE()
-      MESSAGE(FATAL_ERROR "No SSE42 support found. Not building MyRocks")
+      SET(ROCKSDB_BUILD_ARCH "native")
     ENDIF()
+    MESSAGE(STATUS "MyRocks x86_64 build architecture: -march=${ROCKSDB_BUILD_ARCH}")
+  ELSE()
+    MESSAGE(STATUS "Force MyRocks build architecture: -march=${ROCKSDB_BUILD_ARCH}. The created binary may crash on this machine.")
```
Comment (percona-ysorokin) [inline storage/rocksdb/CMakeLists.txt]: `The created binary may crash on this machine.` may be too scary for the user. 
What about
`If the enforced processor architecture is newer than the processor architecture of this build machine, the created binary may crash if executed on this machine`?

File: storage/rocksdb/CMakeLists.txt [hunk 3]
```diff
@@ -161,51 +167,58 @@ ENDIF()
...
+  IF (NOT ROCKSDB_BUILD_ARCH)
+    IF (ROCKSDB_DISABLE_MARCH_NATIVE)
+      SET(ROCKSDB_BUILD_ARCH "armv8-a+crc+crypto")
     ELSE()
-      MESSAGE(FATAL_ERROR "No ARMv8-A+crc+crypto support found. Not building MyRocks")
+      SET(ROCKSDB_BUILD_ARCH "native")
     ENDIF()
+    MESSAGE(STATUS "MyRocks arm64 build architecture: -march=${ROCKSDB_BUILD_ARCH}")
+  ELSE()
+    MESSAGE(STATUS "Force MyRocks build architecture: -march=${ROCKSDB_BUILD_ARCH}. The created binary may crash on this machine.")
```
Comment (percona-ysorokin) [inline storage/rocksdb/CMakeLists.txt]: Likewise

**Review by percona-ysorokin — APPROVED** [review]
LGTM with minor comment

File: storage/rocksdb/CMakeLists.txt (hunk 1 above)
Comment (inikep) [reply storage/rocksdb/CMakeLists.txt:204]: done

File: storage/rocksdb/CMakeLists.txt (hunk 2 above)
Comment (inikep) [reply storage/rocksdb/CMakeLists.txt]: fixed

**Pull request merged**